openpgp/s2k/s2k_config.go

```diff
@@ -35,6 +36,11 @@ type Config struct {
 	// use a value that is at least 65536. See RFC 4880 Section
 	// 3.7.1.3.
 	S2KCount int
+	// When true, allows the S2KMode to be s2k.SaltedS2K.
+	// This is required to ensure the application is aware that a weaker
+	// S2K is being selected, and is only being used when the key has
+	// been randomly generated or a secure S2K has already been applied.
```

Comment:
This comment sounds written with a developer of go-crypto in mind as the intended audience, but the audience should be the users of the library, so you can speak directly to them:
```suggestion
	// Indicates whether the passphrase passed by the application is a
	// high-entropy key (e.g. it's randomly generated or derived from
	// another passphrase using a strong key derivation function).
	// When true, allows the S2KMode to be s2k.SaltedS2K.
	// When the passphrase is not a high-entropy key, using SaltedS2K is
	// insecure, and not allowed by draft-ietf-openpgp-crypto-refresh-08.
```